### PR TITLE
Display challenge data along with renaming function & variable names

### LINKED ISSUE #25: Redesign get_challenge_count to accept both host and participant flags.
The function `get_challenge_count` called when either participant or host is `True` first checks for host and if host is false, checks for participant.

So after activating both flags, only the host challenges are printed. We have to redesign the `get_challenge_count` function to account for this (and possibly rename it too).

#24

### LINKED ISSUE #45: Add test case for testing --host and --participant flag together.

### PR DESCRIPTION
**Pull request by RishabhJain2018**
Fixes #25
Fixes #45 

The major changes that this PR takes into account is:

1. Add commands for displaying the past, ongoing, future, all challenge data.
2. Add commands for displaying the hosted & participated challenges by a user.
Commands: 
```

- evalai challenges
- evalai challenges ongoing
- evalai challenges past
- evalai challenges future
- evalai challenges --host
- evalai challenges --participant
- evalai challenges --host --participant

```
3. Modified the function names along with variables and removed the redundant code.

### CODE REVIEW

**Comment by RishabhJain2018** [comment]
@isht3 Can you please verify on your local that everything works fine if I merge this.

**Comment by RishabhJain2018** [comment]
@varunagrawal Please review the PR.

File: evalai/utils/challenges.py [hunk 1]
```diff
@@ -21,17 +24,21 @@ def print_challenge_table(challenge):
...
-    challenge = "{}{}{}{}".format(title, description, date, br)
+    end_date = "End Date : " + style(challenge["end_date"].split("T")[0], fg="red")
+    end_date = "\n{}\n\n".format(style(end_date, bold=True))
+    challenge = "{}{}{}{}".format(title, description, end_date, br)
     echo(challenge)
 
 
-def get_challenges(url):
+def display_challenges(url):
+
```
Comment (varunagrawal) [inline evalai/utils/challenges.py]: Any reason for this extra line here?

Comment (RishabhJain2018) [reply evalai/utils/challenges.py]: No 😄. Thanks for the catch. I'll fix it and merge the PR.

**Pull request merged**